lib/airbrake/rack/user.rb

```diff
@@ -17,9 +17,15 @@ def self.extract(rack_env)
         end
 
         # Fallback mode (OmniAuth support included). Works only for Rails.
+        user = try_current_user(rack_env)
+        new(user) if user
+      end
+
+      def self.try_current_user(rack_env)
         controller = rack_env['action_controller.instance']
         return unless controller.respond_to?(:current_user)
-        new(controller.current_user) if controller.current_user
+        return unless [-1, 0].include?(controller.method(:current_user).arity)
+        controller.current_user
       end
```

Comment:
Please make this method private ([example](https://github.com/airbrake/airbrake-ruby/blob/a93cd6133f799390086735ddacbee49b8f1286a5/lib/airbrake-ruby/response.rb#L42-L51)).